dask_expr/_merge.py

```diff
@@ -162,12 +162,17 @@ def _lower(self):
         return BlockwiseMerge(left, right, **self.kwargs)
 
     def _simplify_up(self, parent):
-        if isinstance(parent, Projection):
+        if isinstance(parent, (Projection, Index)):
             # Reorder the column projection to
             # occur before the Merge
-            projection = parent.operand("columns")
-            if isinstance(projection, (str, int)):
-                projection = [projection]
+            if isinstance(parent, Index):
+                projection, parent_columns = [], None
```

Comment:
Just to make it easier for others to understand this code in the future, maybe we can we add a comment to clarify that `Index` makes an implicit "empty-column" projection?